src/domain/flow-system/services/src/flow/flow_executor_impl.rs

```diff
@@ -523,17 +525,26 @@ impl MessageConsumerT<TaskProgressMessage> for FlowExecutorImpl {
                     let mut flow = Flow::load(flow_id, flow_event_store.as_ref())
                         .await
                         .int_err()?;
-                    flow.on_task_running(message.event_time, message.task_id)
-                        .int_err()?;
-                    flow.save(flow_event_store.as_ref()).await.int_err()?;
+                    if flow.status() != FlowStatus::Finished {
+                        flow.on_task_running(message.event_time, message.task_id)
+                            .int_err()?;
+                        flow.save(flow_event_store.as_ref()).await.int_err()?;
 
-                    let outbox = target_catalog.get_one::<dyn Outbox>().unwrap();
-                    outbox
-                        .post_message(
-                            MESSAGE_PRODUCER_KAMU_FLOW_PROGRESS_SERVICE,
-                            FlowProgressMessage::running(message.event_time, flow_id),
-                        )
-                        .await?;
+                        let outbox = target_catalog.get_one::<dyn Outbox>().unwrap();
+                        outbox
+                            .post_message(
+                                MESSAGE_PRODUCER_KAMU_FLOW_PROGRESS_SERVICE,
+                                FlowProgressMessage::running(message.event_time, flow_id),
+                            )
+                            .await?;
+                    } else {
+                        tracing::info!(
+                            flow_id=%flow.flow_id,
+                            flow_status=%flow.status(),
+                            task_id=%message.task_id,
```

Comment:
let's respect formatting rules (spaces around `=`) even though `rustfmt` chooses not to mess with the insides of macros

Reply:
Cleaned everywhere, including old code.